packages/optimizer/lib/transformers/RewriteAmpUrls.js

```diff
@@ -49,16 +58,33 @@ class RewriteAmpUrls {
     if (!head) return;
 
     let ampUrlPrefix = params.ampUrlPrefix || AMP_CACHE_HOST;
+    let dynamicAmpUrlPrefix = params.ampUrlPrefix || AMP_CACHE_HOST;
     if (params.ampRuntimeVersion) {
-      ampUrlPrefix = appendRuntimeVersion(ampUrlPrefix, params.ampRuntimeVersion);
+      if (!params.ampUrlPrefix) {
+        ampUrlPrefix = appendRuntimeVersion(ampUrlPrefix, params.ampRuntimeVersion);
+        dynamicAmpUrlPrefix = appendRuntimeVersion(dynamicAmpUrlPrefix, params.ampRuntimeVersion);
+      } else if (params.rewriteDynamicComponents === false) {
+        dynamicAmpUrlPrefix = appendRuntimeVersion(AMP_CACHE_HOST, params.ampRuntimeVersion);
+      }
     }
 
+    // Should be kept up to date with dynamic components listed here:
+    // https://github.com/ampproject/amphtml/blob/master/spec/amp-cache-guidelines.md#guidelines-adding-a-new-cache-to-the-amp-ecosystem
+    const dynamicElements = ['amp-geo'];
```

Comment:
please move this into a constant at the top of the file

Reply:
I added it to `AMP_CONSTANTS`. Good? Bad?